### DISCUSSION TITLE
Best practices for storing sensitive data in Supabase Postgres?

### DISCUSSION THREAD

**Discussion by blakewilson** [supabase/supabase#21073] [Questions]
Hi yall, 

Just trying to wrap my head around best practices with Supabase. From my understanding, Supabase uses a public schema in Postgres that exposes your schema to the REST API and GraphQL endpoints publicly.

I understand that RLS policies can be set so that, for instance, a user can only see their own data and no one elses.

But let's say sensitive information is stored in a public schema's table, something like encrypted third party refresh/access tokens or the like. For example, let's just say a table called `public.tokens`. With an RLS policy, I can make it so a user can only access their own tokens; that makes sense.

But the schema is still public, meaning, bad actors now know the shape of the data/database design.

Am I being paranoid that this schema is public and "not obscure", or is this lack of "security through obscurity" not a problem since the data itself is protected by RLS?

Thanks!

**Comment by blakewilson** [comment]
Related #12439

**Comment by mwoss** [comment]
Your concerns are understandable in my opinion. As far as I know, Supabase uses PostgREST underneath that by default exposes all the tables, views, and stored procedures to the clients for their convenience. You can modify its configuration to disallow this behaviour. Configuration lives inside Postgres so it should be doable. See those materials for more details:

* [PostgREST Schema Isolation docs](https://postgrest.org/en/v12.0/explanations/schema_isolation.html)
* [PostgREST Database Authorization docs](https://postgrest.org/en/stable/explanations/db_authz.html#schemas)
* [PostgREST Configuration docs](https://postgrest.org/en/stable/references/configuration.html#db-schema)
* https://github.com/orgs/supabase/discussions/9464

**Reply by blakewilson** [reply]
Thanks @mwoss. Are you aware if disabling this causes incompatibility with other Supabase functionality? In other words, does Supabase support this or is this a work around and this may cause unintended side effects on the platform?

**Reply by mwoss** [reply]
As far as I know, if you turn off this functionality for end-user roles only (like `authenticator`) everything will work fine :D
